BJS2-41290 deleting expired premiums

ZhubanyshZh opened the pull request.

ZhubanyshZh committed:
feat(41290): improve PremiumRemover Scheduler

change method removeExpiredPremiums to work more better and faster
I use async and batch to this

ZhubanyshZh committed:
test(41290): fix during testing

added test and change something in the scheduler method

ZhubanyshZh committed:
fix(41290): set enableScheduling to App

ZhubanyshZh committed:
fix(41290): added thread pool to method

Shakhaev commented on src/main/resources/application.yaml:
```diff
@@ -51,4 +51,11 @@ minio:
+premium:
+  updater:
+    batch-size: 50
+    cron: 0 0 0 * * *
```
Жубаныш, привет! Насколько я осведомлен, этот cron предполагает удаление раз в день в 00:00, а по условию задачи необходимо реализовать удаление раз в неделю. Думаю, что отработка данной очистки на ежедневной основе может плохо сказываться на производительности приложения.

ZhubanyshZh replied:
Привет, согласен сейчас исправлю

Shakhaev commented on src/main/java/school/faang/user_service/config/scheduler/SchedulerConfig.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.concurrent.Executors;
+
+@Configuration
+public class SchedulerConfig {
```
Жубаныш, а может назвать класс как AsyncPremiumRemoverConfig или ThreadPoolConfig, ведь мы устанавливаем свой кастомный пул потоков для асинхронного удаления и не имеем привязки непосредственно к Scheduled.

ZhubanyshZh committed:
fix(41290): adjust premium remover schedule and config name

- Changed premium remover cron job to run every Sunday at 00:00
- Renamed custom config to AsyncPremiumRemoverConfig for consistency
